packages/pds/src/api/com/atproto/identity/signPlcOperation.ts

```diff
@@ -0,0 +1,46 @@
+import { Server } from '../../../../lexicon'
+import AppContext from '../../../../context'
+import * as plc from '@did-plc/lib'
+import { check } from '@atproto/common'
+import { InvalidRequestError } from '@atproto/xrpc-server'
+
+export default function (server: Server, ctx: AppContext) {
+  server.com.atproto.identity.signPlcOperation({
+    auth: ctx.authVerifier.accessNotAppPassword,
+    handler: async ({ auth, input }) => {
+      const did = auth.credentials.did
+      const { token } = input.body
+      if (!token) {
+        throw new InvalidRequestError(
+          'email confirmation token required to sign PLC operations',
+        )
+      }
+      await ctx.services
+        .account(ctx.db)
+        .assertValidToken(did, 'plc_operation', token)
+
+      const lastOp = await ctx.plcClient.getLastOp(did)
+      if (check.is(lastOp, plc.def.tombstone)) {
+        throw new InvalidRequestError('Did is tombstoned')
+      }
+      const operation = await plc.createUpdateOp(
+        lastOp,
+        ctx.plcRotationKey,
+        (lastOp) => ({
+          ...lastOp,
+          rotationKeys: input.body.rotationKeys ?? lastOp.rotationKeys,
```

Comment:
Just making sure it doesn't make sense to do the `lastOp` tombstone check inside the callback here.

Reply:
yup checks in the right spot, but i updated the code to use `prev` in the callback to avoid confusion